src/providers/originProvider.ts

```diff
@@ -0,0 +1,105 @@
+import { CACHE_HEADERS } from '../constants/cache';
+import { Context } from '../context';
+import {
+  GetFileOptions,
+  GetFileResult,
+  HeadFileResult,
+  HttpResponseHeaders,
+  Provider,
+  ReadDirectoryResult,
+} from './provider';
+
+type OriginProviderCtorOptions = {
+  ctx: Context;
+};
+
+/**
+ * Serves assets from origin.nodejs.org, used as a fallback for if R2 fails.
+ */
+export class OriginProvider implements Provider {
+  private ctx: Context;
+
+  constructor({ ctx }: OriginProviderCtorOptions) {
+    this.ctx = ctx;
+  }
+
+  async headFile(path: string): Promise<HeadFileResult | undefined> {
```

Comment:
What the method `headFile` does? Just to fulfil a HEAD request?

Reply:
Yeah, it's there just to perform HEADs on file paths. Separated it from getFile so the code paths can be cleaner and more straight forward (without a bunch of `if (request.method === 'GET') {...}` checks that are present now)